### PR TITLE
Streamline point and tetra interp method

### PR DESCRIPTION
**Pull request by mforets**
Prep for https://github.com/ONSAS/ONSAS.jl/issues/244.

The `Point` struct is now simply a type alias with `SVector` (one can still `Point(vec...)` for arbitrary arrays).

The tetra intep method now uses static arrays and returns a static vector (point). Such type stability is useful for subsequent computations.

### CODE REVIEW

File: test/elements.jl [hunk 1]
```diff
@@ -282,15 +282,16 @@ n₄ = Node(2.0, 0.0, 1.0,
     # The interpolation for a linear scalar field shloud be exact 
     scalar_linear_field(x, y, z) = 10x + 20y + 30z + 40
     sol_at_tetra_nodes = [scalar_linear_field(coordinates(n)...) for n in nodes(tetra)]
-    p = [0.5, 0.5, 0.5]
+    p = Point(0.5, 0.5, 0.5)
     exact_solution = scalar_linear_field(p...)
+    @show tetra
+    @show p
```
Comment (mforets) [inline test/elements.jl]:
```suggestion
```

File: src/Elements/Tetrahedron.jl [hunk 2]
```diff
@@ -206,38 +205,41 @@ function _shape_functions_derivatives(::Tetrahedron, order=1)
...
     return 𝑀
 end
 
 "Return the interpolation weights of a point `p` in a `Tetrahedron` element `t`."
-function weights(t::Tetrahedron{3,T}, p::Point{dim,P}) where {T<:Real,dim,P<:Real}
-    @assert length(p) == 3 "The point $p must be a 3D vector."
-    return _interpolation_matrix(t) * [1, p...]
+function weights(t::Tetrahedron{3}, p::Point{dim}) where {dim}
+    @assert dim == 3 "The point $p must be a 3D vector."
+    return interpolation_matrix(t) * Point(1, p...)
```
Comment (mvanzulli) [inline src/Elements/Tetrahedron.jl]:
```suggestion
    interpolation_matrix(t) * Point(1, p...)
```

File: src/Elements/Tetrahedron.jl [hunk 3]
```diff
@@ -206,38 +205,41 @@ function _shape_functions_derivatives(::Tetrahedron, order=1)
...
     end
     return 𝑀
 end
 
 "Return the interpolation weights of a point `p` in a `Tetrahedron` element `t`."
-function weights(t::Tetrahedron{3,T}, p::Point{dim,P}) where {T<:Real,dim,P<:Real}
-    @assert length(p) == 3 "The point $p must be a 3D vector."
-    return _interpolation_matrix(t) * [1, p...]
+function weights(t::Tetrahedron{3}, p::Point{dim}) where {dim}
+    @assert dim == 3 "The point $p must be a 3D vector."
```
Comment (mvanzulli) [inline src/Elements/Tetrahedron.jl]: Maybe we can delete the `assert` and just use `p::Point{3}`

**Review by mvanzulli — CHANGES_REQUESTED** [review]

File: src/Elements/Tetrahedron.jl (hunk 3 above)
Comment (mforets) [reply src/Elements/Tetrahedron.jl]: yes

**Review by mvanzulli — APPROVED** [review]

**Pull request merged**